test/bin/scenario.sh

```diff
@@ -227,6 +269,8 @@ prepare_kickstart() {
     local -r vm_hostname="${full_vmname/./-}"
     local -r hostname=$(hostname)
 
+    validate_full_vm_name $full_vmname
```

Comment:
I think we want to validate `vm_hostname` since it'll be used for hostname.
```suggestion
    validate_full_vm_name "${vm_hostname}"
```